rascal-lsp/src/main/rascal/lsp/lang/rascal/lsp/IDECheckerWrapper.rsc

```diff
@@ -100,15 +100,18 @@ map[loc, set[Message]] checkFile(loc l, set[loc] workspaceFolders, start[Module]
             step3("Checked module in `<project.file>`", 1);
             pcfg = getPathConfig(project);
             checkOutdatedPathConfig(pcfg);
-            msgs += check([*modulesPerProject[project]], rascalCompilerConfig(pcfg));
+            modulesToCheck = calculateOutdated(modulesPerProject[project], pcfg);
+            if (modulesToCheck != []) {
+                msgs += check(modulesToCheck, rascalCompilerConfig(pcfg));
+            }
         }
         return true;
     }, totalWork=size(upstreamDependencies));
 
     step("Checking module <l>", 1);
     pcfg = getPathConfig(initialProject);
     checkOutdatedPathConfig(pcfg);
-    msgs += check([l], rascalCompilerConfig(pcfg));
+    msgs += check(calculateOutdated(modulesPerProject[initialProject] + l, pcfg), rascalCompilerConfig(pcfg));
```

Comment:
Doesn't the type checker handle files within the project already?

Reply:
The assumption is that the typechecker only gets a list of *changed* modules to check.

Reply:
But if I would pass in the request to check `A` and it imports `B`, would it not figure out that `B` is newer than its tpl?

Reply:
Okay, I'll keep the behavior as here in the PR, just so that it more closely follows the maven plugin. With the understanding that the typechecker should not only lean on that, but keep its checking logic for tpl invalidation.